src/OpenLoco/src/Ui/Window.cpp

```diff
@@ -674,7 +674,12 @@ namespace OpenLoco::Ui
 
     bool Window::viewportIsFocusedOnEntity(EntityId targetEntity) const
     {
-        if (targetEntity == EntityId::null || viewports[0] == nullptr || savedView.isEmpty())
+        if (viewports[0] == nullptr || savedView.isEmpty())
+        {
+            return false;
+        }
+
+        if (targetEntity != EntityId::null && viewportConfigurations[0].viewportTargetSprite == EntityId::null)
```

Comment:
```suggestion
        // If the argument provided is not EntityId::null check for a match.
        if (targetEntity != EntityId::null && viewportConfigurations[0].viewportTargetSprite != targetEntity)
```
And the next check should be just look to see if the assigned is non-null, I think this is a bit easier to grasp.